### DISCUSSION TITLE
Find the model spatial structure

### DISCUSSION THREAD

**Discussion by mitch46syd** [IfcOpenShell/IfcOpenShell#3962] [General]
I'm writing a little tool to open an ifc file, show the spatial structure and any geo-referencing .
I've used the method `decomposition = ifcopenshell.util.element.get_decomposition(element)` 
and get the data:
`#11=IfcProject('04hBSiSzPC5fsYypg9Mn7y',#10,'E4 Förbifart Stockholm (Program)',$,$,$,$,$,$)
[#4=IfcSite('3aVxnycuTFhuaONkykPgUe',#10,'Skärholmen (Delomrade)',$,$,#3,$,$,$,$,$,$,$,$), #13=IfcBridge('3_8eV586P0muvFD5A3Fzqn',#10,$,$,$,$,$,$,$,$), #26=IfcFacilityPart('2r6fGfMerF2OJ6dLEL409U',#10,'överbyggnad (31--)',$,$,#25,$,$,$,IfcBridgePartTypeEnum(.SUPERSTRUCTURE.)), #31=IfcFacilityPart('0fTB94WKX2dgrmqTEjB5J6',#10,'underbyggnad (12--)',$,$,#30,$,$,$,IfcBridgePartTypeEnum(.SUBSTRUCTURE.)), #35=IfcFacilityPart('0fTB94WKX2dgrmqTEjB5J4',#10,'undergrund (11--)',$,$,#34,$,$,$,IfcBridgePartTypeEnum(.FOUNDATION.)), #39=IfcFacilityPart('0i3NlGsbb3XRylMEwXSI1b',#10,'Anläggningskomplettering (..--)',$,$,#38,$,$,$,IfcFacilityPartCommonTypeEnum(.BELOWGROUND.))]`

but  I cannot understand how to extract the information. Any insights would be most appreciated

**Comment by Moult** [comment]
What information are you trying to extract?

**Reply by mitch46syd** [reply]
Hi Dion - the list of entities. Embarrassingly I just tried listing  the entities and that's 90% of solution. 
Tks for immediate response. You could delete this ?

**Reply by Moult** [reply]
Uh, ok :)
